File: src/main/java/com/example/app/utils/model/entities/Salt.java
package com.example.app.utils.model.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Salt {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    private String salt1;
    private String salt2;

    private String owner;

    public Salt(String salt1, String salt2, String owner){
        this.salt1 = salt1;
        this.salt2 = salt2;
        this.owner = owner;
    }

}

File: src/main/kotlin/ru/kaplaan/domain/entity/Salt.kt
package ru.kaplaan.domain.entity

import jakarta.persistence.*
import jakarta.validation.constraints.NotBlank
import org.springframework.validation.annotation.Validated

@Entity
data class Salt(
    var salt1: String,
    var salt2: String,
    var owner: String,
){
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private var id: Long? = null

    constructor(id: Long, salt1: String, salt2: String, owner: String): this(salt1, salt2, owner){
        this.id = id
    }
}
